pkg/sql/ir_generator.go

```diff
@@ -538,27 +564,43 @@ func parseNumericColumn(el *parser.ExprList) (*ir.NumericColumn, error) {
 }
 
 func parseBucketColumn(el *parser.ExprList) (*ir.BucketColumn, error) {
-	help := "BUCKET(NUMERIC(...), BOUNDARIES)"
+	help := "BUCKET([NUMERIC(...)|col_name], BOUNDARIES)"
 	if len(*el) != 3 {
 		return nil, fmt.Errorf("bad BUCKET expression format: %s, should be like: %s", *el, help)
 	}
 
 	sourceExprList := (*el)[1]
 	boundariesExprList := (*el)[2]
-	if sourceExprList.Type != 0 {
-		return nil, fmt.Errorf("key of BUCKET must be NUMERIC, which is %v", sourceExprList)
-	}
-	source, err := parseFeatureColumn(&sourceExprList.Sexp)
-	if err != nil {
-		return nil, err
-	}
-	if _, ok := source.(*ir.NumericColumn); !ok {
-		return nil, fmt.Errorf("key of BUCKET must be NUMERIC, which is %s", source)
+
+	var source ir.FeatureColumn
+	var err error
+
+	if sourceExprList.Sexp == nil {
```

Comment:
Better to use `sourceExprList.Type` to determine whether the subexpression is a string or an expression.

Reply:
OK. I will fix it in next PR asap.

Reply:
OK. I will fix it in next PR asap.